cmd/glasskube/cmd/bootstrap_git.go

```diff
@@ -69,8 +84,18 @@ var bootstrapGitCmd = &cobra.Command{
 
 		// regularly bootstrap in the cluster
 		// without a git client, this means it only supports GitHub for now
-		basePath := strings.ReplaceAll(bootstrapGitCmdOptions.url, "github.com", "raw.githubusercontent.com")
-		basePath = basePath + "/main"
+		host := "raw.githubusercontent.com"
+		if bootstrapGitCmdOptions.token != "" {
+			// might also be GitHub specific? a more proper way for the future could be to send it via the
+			// Authorization header instead, but that would also have to be supported by the BootstrapClient
+			host = bootstrapGitCmdOptions.token + "@" + host
```

Comment:
I am somehow amazed that this even works, lol

But is the username even needed, apart from saving it to the secret below?

Reply:
I thought the username can be removed entirely, but on closer inspection it appears that it is needed for argo-cd to access the repo (?) 

If it can not be removed feel free to mark this conversation as resolved